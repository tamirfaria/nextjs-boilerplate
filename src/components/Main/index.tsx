const Main = () => {
  return (
    <div>
      <h1>Main Test</h1>
    </div>
  )
}

export default Main
