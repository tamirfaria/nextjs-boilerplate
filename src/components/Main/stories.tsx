import { ComponentMeta, ComponentStory } from '@storybook/react'

import Main from '.'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'Default',
    description: 'Description defaultn'
  }
} as ComponentMeta<typeof Main>

export const Default: ComponentStory<typeof Main> = (args) => <Main {...args} />

export const Primary: ComponentStory<typeof Main> = (args) => <Main {...args} />
Primary.args = { title: 'Primary', description: 'Primary description' }

export const Secondary: ComponentStory<typeof Main> = (args) => (
  <Main {...args} />
)

Secondary.args = { title: 'Secondary', description: 'Secondary description' }
