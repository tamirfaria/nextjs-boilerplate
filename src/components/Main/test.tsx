import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main/>', () => {
  it('should render Main without break', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /react avanÃ§ado/i })
    ).toBeInTheDocument()
    expect(
      screen.getByRole('heading', {
        name: /typescript, reactjs, nextjs e styled components/i
      })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the background color correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
