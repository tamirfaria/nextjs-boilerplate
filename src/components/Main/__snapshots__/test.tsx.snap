// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> should render Main without break 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #06092b;
  color: #fff;
  width: 100vw;
  height: 100vh;
  padding: 3rem;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

.c4 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

<main
  class="c0"
>
  <img
    alt="Imagem de um átomo com o texto React Avançado ao lado"
    class="c1"
    src="/img/logo.svg"
  />
  <h1
    class="c2"
  >
    React Avançado
  </h1>
  <h2
    class="c3"
  >
    Typescript, ReactJS, NextJS e Styled Components
  </h2>
  <img
    alt="Um desenvolvedor sentado em uma cadeira de frente para um monitor"
    class="c4"
    src="/img/hero-illustration.svg"
  />
</main>
`
